### PR TITLE
dev-python/patroni: initial commit

### PR DESCRIPTION
**Pull request by peterbangert**
Currently is downloadable and repoman approves.

However I believe I should include zookeeper as a dependecny since aleksi wants tthat specifically.

### CODE REVIEW

**Comment by einhverfr** [comment]
Not quite sure what Repoman is complaining about:

```
  dependency.bad [fatal]        39

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/desktop)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/desktop/gnome)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/desktop/gnome/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/desktop/plasma)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/desktop/plasma/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/developer)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/hardened)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/hardened/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/no-multilib)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/no-multilib/hardened)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/no-multilib/hardened/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/desktop)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/desktop/gnome)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/desktop/gnome/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/desktop/plasma)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/desktop/plasma/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/developer)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/hardened)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/hardened/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/no-multilib)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/no-multilib/hardened)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/no-multilib/hardened/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~amd64(default/linux/amd64/17.1/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/desktop)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/desktop/gnome)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/desktop/gnome/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/desktop/plasma)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/desktop/plasma/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/developer)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/hardened)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/hardened/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/selinux)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']

   dev-python/patroni/patroni-1.6.5.ebuild: DEPEND: ~x86(default/linux/x86/17.0/systemd)

[     'dev-python/flake8[python_targets_python2_7(-)?,python_targets_python3_6(-)?,python_targets_python3_7(-)?,-python_single_target_python2_7(-),-python_single_target_python3_6(-),-python_single_target_python3_7(-)]']
```

**Comment by einhverfr** [comment]
(TBH, wondering if this is a travis/repoman issue)

**Comment by peterbangert** [comment]
@einhverfr perhaps i should package flake8 as well. 

im not able to diagnose the screams

**Comment by einhverfr** [comment]
Also on zookeeper as a dependency, it does not need to be on the same host, so I don;t think that is required.

**Comment by einhverfr** [comment]
(we need the zookeeper client libraries maybe optionally)

**Comment by patricklauer** [comment]
> dev-python/flake8[python_targets_python2_7]

py2 delendam est!

(Try not to add any dependencies on py2 versions, it's STRONGLY DISCOURAGED)

File: dev-python/patroni/patroni-1.6.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6,3_7} )
```
Comment (patricklauer) [inline dev-python/patroni/patroni-1.6.5.ebuild:6]: python{2_7,3_4,3_5} are mostly useless, best to remove

File: dev-python/patroni/patroni-1.6.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	dev-python/psycopg:2[${PYTHON_USEDEP}]
+	dev-python/flake8[${PYTHON_USEDEP}]
```
Comment (patricklauer) [inline dev-python/patroni/patroni-1.6.5.ebuild:21]: I'd expect flake8 to be a test dep only?

**Comment by peterbangert** [comment]
ok im thinking everythings looking much better now.

**Pull request merged**